lib/defender/aws_defender.c

```diff
@@ -276,11 +281,41 @@ static DefenderState_t prvStateSubscribe( void )
     return eDefenderStateSubscribeMqttSuccess;
 }
 
+static char * prvGetDefenderMqttTopicString( char * pcTopicTemplate )
+{
+    char * pcTopic = NULL;
+    uint32_t ulTopicLength = 0;
+    uint32_t ulCharsWritten = 0;
+
+    /* Determine the maximum resulting string length including NULL terminator.
+     * The input parameter is assumed to include a %s for inserting the Thing
+     * name.*/
+    ulTopicLength = strlen( pcTopicTemplate ) +
+                    strlen( clientcredentialIOT_THING_NAME ) - 1;
+
+    /* Allocate memory. */
+    pcTopic = pvPortMalloc( ulTopicLength );
+
+    if( NULL != pcTopic )
+    {
+        /* Build the topic string including the device name. */
+        ulCharsWritten = snprintf( pcTopic,
+                                   ulTopicLength,
+                                   pcTopicTemplate,
+                                   clientcredentialIOT_THING_NAME );
+
+        if( ulCharsWritten >= ulTopicLength )
```

Comment:
If snprintf fails it returns a negative number.

Should we just check if (ulCharsWritten != ulTopicLength) --> https://github.com/aws/amazon-freertos/pull/635